.github/workflows/black.yml

```diff
@@ -0,0 +1,19 @@
+name: Black
+
+on: [push, pull_request]
+
+jobs: 
+  Black:
+    runs-on: ubuntu-20.4
+    container: fedora:32
+    steps:
+      - name: Install Pipenv and Git
+        run: dnf install -y pipenv git 
+      - name: Checkout code
+        uses: actions/checkout@v2
+        name: Setup environment
+        run: pipenv sync --dev 
+        name: Run Black
+        uses: psf/black@stable
+        with: 
+        args: ". --check"
```

Comment:
If you want to reduce the string single or double quote changes, you can add `--skip-string-normalization`.

Reply:
am I adding this in the 'args' section? And should I do that? what's the best practice?

Reply:
It is an argument of `black`, so I do not understand your question.

Regarding if this option is appropriate or not is debatable and is context dependent.
Personally, I like single quotes because I find it cleaner (less pixels), so in the projects I was involved I did used this option. On the other hand, it makes it harder to enforce a format, as you will need to comment about it in reviews to keep it in this standard.
In this case, you can even do it in two phases: Do not enforce it at start, and in a following change do enforce it. But as the codebase is not that big, I think you do not need this steps, just decide what you prefer.

Bottom line, it is up to you. Just reason for your action and decision in the commit so it will be recorded for the future.

Reply:
I was just making sure since I'm still not familiar entirely with YML files. Regarding my decision, I definitely prefer single quotes in most places it already has been, also as it was a default in some of Django default files, I see this formatting files commit as too vast and unnecessary at this point, I will make the changes and commit the formatted files after adding this argument, and later we can decide publicly where we do prefer double quotes if any in our code without using the formatting of Black.